src/main/java/com/denizenscript/depenizen/bukkit/objects/breweryx/BreweryRecipeTag.java

```diff
@@ -0,0 +1,349 @@
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.bRecipe.getRecipeName());
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.ingredients>
+        // @returns ListTag(ItemTag)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns a ListTag of ItemTags that are the ingredients of the recipe.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "ingredients", (attribute, object) -> {
+            ListTag ingredients = new ListTag();
+            for (RecipeItem recipeItem : object.bRecipe.getIngredients()) {
+                if (recipeItem.getMaterials() == null) {
+                    continue;
+                }
+                for (Material material : recipeItem.getMaterials()) {
+                    ingredients.addObject(new ItemTag(material, recipeItem.getAmount()));
+                }
+            }
+            return ingredients;
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.difficulty>
+        // @returns ElementTag(Number)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the difficulty of the recipe.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "difficulty", (attribute, object) -> {
+            return new ElementTag(object.bRecipe.getDifficulty());
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.cooking_time>
+        // @returns ElementTag(Number)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the cooking time of the recipe.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "cooking_time", (attribute, object) -> {
+            return new ElementTag(object.bRecipe.getCookingTime());
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.distill_runs>
+        // @returns ElementTag(Number)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the distill runs of the recipe
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "distill_runs", (attribute, object) -> {
+            return new ElementTag(object.bRecipe.getDistillRuns());
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.distill_time>
+        // @returns ElementTag(Number)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the amount of time each distill run takes.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "distill_time", (attribute, object) -> {
+            return new ElementTag(object.bRecipe.getDistillTime());
+        });
+
+        // <--[tag]
+        // @attribute <BRecipeTag.wood>
+        // @returns ElementTag(Number)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the type of wood used in the recipe (by number, Ex: 0 = Any, 1 = Oak).
```

Comment:
That's a very confusing system - why is there not names?

Reply:
Written by the original author(s). Wood type are represented in the config and throughout the plugin as numbers

Reply:
BreweryX now uses names in addition to the original numbering system. Returns an ElementTag of an enum now